src/main/java/jdbchomework/dao/jdbc/DevelopersJdbcDao.java

```diff
@@ -2,11 +2,42 @@
 
 import jdbchomework.dao.model.DevelopersDao;
 import jdbchomework.entity.Developer;
-import org.hibernate.SessionFactory;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
 
 public class DevelopersJdbcDao extends AbstractDao<Developer> implements DevelopersDao {
 
-    public DevelopersJdbcDao(SessionFactory sessionFactory, String entity, Class<Developer> clazz) {
-        super(sessionFactory, entity, clazz);
+    private static org.slf4j.Logger log = LoggerFactory.getLogger(DevelopersJdbcDao.class);
+
+    public DevelopersJdbcDao(Connection connection, String table, String column) {
+        super(connection, table, column);
+    }
+
+    @Override
+    public boolean updateById(long id, Developer toUpdate) {
+        boolean result = false;
+        try (PreparedStatement statement = connection
+                .prepareStatement("UPDATE developers SET name = ?, salary = ? WHERE developer_id =?;")) {
+            statement.setString(1, toUpdate.getName());
+            statement.setInt(2, toUpdate.getSalary());
+            statement.setLong(3, id);
+            if (statement.executeUpdate() > 0) {
+                result = true;
+            }
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException("Cannot connect to DB", e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)